earthly/mdlint/Earthfile

```diff
@@ -0,0 +1,61 @@
+VERSION 0.7
+
+MDLINT:
+    # Linting is done with MarkdownLint CLI2
+    # See: https://github.com/DavidAnson/markdownlint-cli2
+    # We use a special container which includes all rule extensions to markdownlint
+    # Notably, we rely on the `max-one-sentence-per-line` rule which helps keep
+    # diffs of markdown files small and legible.  This rule is only currently
+    # in the `next` branch of `markdownlint`.
+    # Container is built by:
+    #   https://github.com/DavidAnson/markdownlint-cli2/blob/next/docker/Dockerfile-rules
+    COMMAND
+
+    # Directory we need to run lint checks against.
+    ARG --required src
+
+    # Optional `fix` argument.  Must be either not set, or set to `--fix` anything else is an error.
+    ARG fix
+
+    # Unlikely this ever needs to be changed.
+    ARG cfg_file=.markdownlint-cli2.jsonc
+
+    # Ensure the `fix` argument only contains valid values.
+    RUN if [[ "${fix}" != "--fix" && "${fix}" != "" ]]; then \
+        echo "Invalid value for fix argument: ${fix}. It should be either --fix or not set." >&2; \
+        exit 1; \
+    fi
+
+    # Ensure the path to be checked has a mdlint config file.
+    RUN --no-cache if [[ ! -f $src/$cfg_file ]]; then \
+        (>&2 echo "$src/$cfg_file does not exist. Can not run mdLint."); \
+        exit 1; \
+    fi
+
+    # Status line for what we are about to do.
+    RUN --no-cache echo Linting Markdown Recursively from: $src
+
+    # Run the linter with the given arguments, and recursively check all markdown files.
+    # The directory to be checked `MUST` have a `.markdownlint-sli2.jsonc` file.
+    # cspell: words davidanson
+    WITH DOCKER --pull davidanson/markdownlint-cli2-rules:next
```

Comment:
See previous comment about using `WITH DOCKER`.

Reply:
Same reply, ultra-low priority with no clear return on investment vs the cost of reimplementing something that already works.  Especially given the ongoing maintenance needed to track upstream changes as and when they occur.

Mdlint, and CSpell are not just a single file that's run and done. They package with plugins and data files and what not.  All of which we would be taking on the burden of maintaining.  Until I see this actually cause an issue with a reproducable example I can't see any value in it.

I also don't see any reason it can't work OK if `FROM` can happily pull a Container published to a container repo?

We could go:

```Earthly
my-mdlint:
  FROM davidanson/markdownlint-cli2-rules:next
```

And that's no different from pulling alpine or debian.  The big difference here is we don;t want to customize it at all.

Reply:
Again,  is it your suggestion to do this an alternative way, or to rebuild the markdownlint container ourselves?